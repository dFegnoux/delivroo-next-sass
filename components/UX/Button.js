import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { ThemeContext } from "../../contexts/theme-context";
import Link from "next/link";
import "./button.scss";

class Button extends PureComponent {
  static propTypes = {
    handleClick: PropTypes.func,
    href: PropTypes.string,
    label: PropTypes.string,
    disabled: PropTypes.bool,
    className: PropTypes.string
  };

  static defaultProps = {
    label: "",
    fullWidth: false,
    className: ""
  };

  render() {
    const { handleClick, href, label, disabled, className } = this.props;
    const theme = this.context;
    const btnClass = `btn ${theme} ${className}`;
    if (typeof handleClick === "function") {
      return (
        <button className={btnClass} onClick={handleClick} disabled={disabled}>
          {label}
        </button>
      );
    }
    return (
      <Link href={href}>
        <button className={btnClass} disabled={disabled}>
          {label}
        </button>
      </Link>
    );
  }
}

Button.contextType = ThemeContext;

export default Button;
