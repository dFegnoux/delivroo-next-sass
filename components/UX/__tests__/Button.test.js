import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Button from "../Button";

describe("Buttons snapshots", () => {
  it("Should render a buttn when handleclick is setted", () => {
    const handleClick = jest.fn();
    const enabledBtn = renderer
      .create(
        <Button handleClick={handleClick} label="toto" className="delivroo" />
      )
      .toJSON();
    expect(enabledBtn).toMatchSnapshot();

    const disabledBtn = renderer
      .create(
        <Button
          handleClick={handleClick}
          label="toto"
          className="delivroo"
          disabled
        />
      )
      .toJSON();
    expect(disabledBtn).toMatchSnapshot();
  });

  it("Should render a Link when href is setted", () => {
    const enabledLink = renderer.create(<Button href="/" />).toJSON();
    expect(enabledLink).toMatchSnapshot();

    const disabledLink = renderer
      .create(<Button href="/" label="toto" className="delivroo" disabled />)
      .toJSON();
    expect(disabledLink).toMatchSnapshot();
  });
});

describe("Handle click button behavior", () => {
  const handleClick = jest.fn();
  const container = shallow(<Button handleClick={handleClick} />);

  it("Should render a <button> when handleclick is setted", () => {
    expect(container.find("button").length).toBe(1);
  });
  it("Should fired the handleClick function on click", () => {
    container.simulate("click");
    expect(handleClick).toHaveBeenCalled();
  });
});

describe("Link button behavior", () => {
  it("Should render a Link when href is setted", () => {
    const container = shallow(<Button href="/" />);
    expect(container.find("button").length).toBe(1);
  });
});
