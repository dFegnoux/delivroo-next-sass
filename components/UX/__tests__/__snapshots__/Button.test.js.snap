// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons snapshots Should render a Link when href is setted 1`] = `
<button
  className="btn [object Object] "
  onClick={[Function]}
>
  
</button>
`;

exports[`Buttons snapshots Should render a Link when href is setted 2`] = `
<button
  className="btn [object Object] delivroo"
  disabled={true}
  onClick={[Function]}
>
  toto
</button>
`;

exports[`Buttons snapshots Should render a buttn when handleclick is setted 1`] = `
<button
  className="btn [object Object] delivroo"
  onClick={[MockFunction]}
>
  toto
</button>
`;

exports[`Buttons snapshots Should render a buttn when handleclick is setted 2`] = `
<button
  className="btn [object Object] delivroo"
  disabled={true}
  onClick={[MockFunction]}
>
  toto
</button>
`;
